### PR TITLE
Use networkVersion for network state; chainId for signing transactions

### PR DESCRIPTION
**Pull request by rekmarks**
Changes were recently made to ensure that chain IDs are used to sign transactions. It caused the `networkController.networkState` to use hexadecimal `chainId` values instead of decimal `networkVersion` values. This created unintended downstream effects.

This PR ensures the we preserve existing behavior for `NetworkController.networkState`/`state.metamask.network` to the greatest extent possible, while using explicitly specified `chainId` values for signing transactions.

### CODE REVIEW

File: app/scripts/controllers/network/network.js [hunk 1]
```diff
@@ -145,11 +143,16 @@ export default class NetworkController extends EventEmitter {
         }
 
         // Now we set the network state to the chainId computed earlier
```
Comment (Gudahtt) [inline app/scripts/controllers/network/network.js:145]: Nit: this comment is no longer accurate

Comment (rekmarks) [reply app/scripts/controllers/network/network.js]: 54ddbcc

File: app/scripts/controllers/network/network.js [hunk 2]
```diff
@@ -136,20 +134,24 @@ export default class NetworkController extends EventEmitter {
...
       const currentNetwork = this.getNetworkState()
       if (initialNetwork === currentNetwork) {
         if (err) {
           this.setNetworkState('loading')
           return
         }
 
-        // Now we set the network state to the chainId computed earlier
-        this.setNetworkState(chainId)
+        this.setNetworkState(networkVersion)
```
Comment (Gudahtt) [inline app/scripts/controllers/network/network.js:145]: This still constitutes a change in behavior, as beforehand we used the network version for the built-in Infura networks, but we used the `chainId` field for custom RPC networks. This is relied upon in various places in our codebase, such as for address book entries.

Comment (rekmarks) [reply app/scripts/controllers/network/network.js]: This was fixed in 8b5b254

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**